VL.Data/Services/BookService.cs

```diff
@@ -0,0 +1,63 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using VL.Shared.Data;
+using VL.Shared.Interfaces;
+using VL.Shared.Model;
+
+namespace VL.Shared.Services
+{
+    public class BookService : IBookService
+    {
+        private readonly ApplicationDbContext _applicationDbContext;
+        private readonly IMapper _mapper;
+
+        public BookService(ApplicationDbContext applicationDbContext, IMapper mapper)
+        {
+            _applicationDbContext = applicationDbContext;
+            _mapper = mapper;
+        }
+
+        public async Task<IEnumerable<BookDto>> GetBooksAsync()
+        {
+            var bookList = await _applicationDbContext.Book
+                .ToListAsync();
+            var mappedBookList = bookList.Select(bookList => _mapper.Map<BookDto>(bookList));
+            return mappedBookList;
+        }
+
+        public async Task<BookDto?> GetBookAsync(int id)
+        {
+            var book = await _applicationDbContext.Book.FindAsync(id);
+            if (book == null) return null;
+            var mappedBook = _mapper.Map<BookDto>(book);
```

Comment:
Don't invent variables, just use the type (especially when using var)

`mappedBook` -> `bookDto``